### PR TITLE
herdtools7 release, follow LKMM changes.

### PR DESCRIPTION
**Pull request by maranget**
This intermediate release is necessary to follow changes in the Linux Kernel Memory Model (LKMM).

Thanks

### CODE REVIEW

**Comment by mseri** [comment]
There is a strange failure:
```
#=== ERROR while compiling herdtools7.7.56.3 ==================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/herd/herdtools7/archive/refs/tags/7.56.3.tar.gz)
# path                 ~/.opam/4.11/.opam-switch/build/herdtools7.7.56.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build sh ./dune-build.sh /home/opam/.opam/4.11
# exit-code            1
# env-file             ~/.opam/log/herdtools7-7-9e2a73.env
# output-file          ~/.opam/log/herdtools7-7-9e2a73.out
### output ###
# + ./version-gen.sh /home/opam/.opam/4.11
# + dune build --profile release @all
# Module Parser in directory _build/default/asllib depends on Asllib.
# This doesn't make sense to me.
# 
# Asllib is the main module of the library and is the only module exposed 
# outside of the library. Consequently, it should be the one depending 
# on all the other modules in the library.
```

**Comment by maranget** [comment]
Hi @mseri and @kit-ty-kate, I have changed the constraints so as to require a more recent menhir. This change has fixed the problem. However there are still some failures. I do not know how to fix the first four that are related to ocamlopt not being available on some platforms.

File: packages/herdtools7/herdtools7.7.56.3/opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+build: ["sh" "./dune-build.sh" "%{prefix}%"]
+install: ["sh" "./dune-install.sh" "%{prefix}%"]
+# @todo Add "build-doc" field
+# @todo Add "build-test" field
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune"  {>= "1.4" }
+  "menhir" {>= "20200123"}
+]
+url {
```
Comment (mseri) [inline packages/herdtools7/herdtools7.7.56.3/opam:22]:
```suggestion
conflicts: ["ocaml-option-bytecode-only"]
url {
```

**Comment by maranget** [comment]
Thanks @mseri , I assume that the CI no longer attempts compiling the package on systems where ocamlopt is not available. This is fine by me. However there are new failure, but they seem to be related to the CI process. Is there anything I can do?

**Comment by mseri** [comment]
You can ignore the infrastructural failures, this can be merged. Srry for the wait

**Comment by maranget** [comment]
Thanks @mseri.

**Pull request merged**